app/mailers/application_mailer.rb

```diff
@@ -8,4 +8,27 @@ class ApplicationMailer < Mail::Notify::Mailer
       "GOVUK_NOTIFY_TEMPLATE_ID_APPLICATION",
       "95adafaf-0920-4623-bddc-340853c047af",
     )
+
+  rescue_from Notifications::Client::RequestError do
+    # WARNING: this needs to be a block, otherwise the exception will not be
+    # re-raised and we will not be notified via Sentry, and the job will not retry.
+    #
+    # @see https://github.com/rails/rails/issues/39018
+    if respond_to?(:headers)
+      email_address = headers["To"].value
+      mailer_action_method = action_name
+      mailer_class = mailer_name
+      MailDeliveryFailure.create!(
+        email_address:,
+        mailer_action_method:,
+        mailer_class:,
+      )
+    end
+
+    raise
+  end
+
+  def notify_email(headers)
```

Comment:
Do we need this method here? It looks like a convenience method for calling `view_mail` with the template ID already provided, but it's not used anywhere.

Reply:
It was originally going to add the mailer_action_method and mailer_class to the header but that was done automatically. However It's still used in the application mailer spec on line 15 to set some dummy info.